File: ProblemSolve/461.cpp
/********************
 * Hamming Distnace *
 ********************/

class Solution {
public:
    int hammingDistance(int x, int y) {
        //서로 다른 비트만 1로 만듦
        int check = x ^ y;
        int distance = 0;
                    
        //1인 비트 체크
        while(check) {
            if(check & 1){
                distance++;
            }
            check = check >> 1;
        }
    
        return distance;
    }
};

File: ProblemSolve/461.cs
/********************
 * Hamming Distance *
 ********************/

public class Solution {
    public int HammingDistance(int x, int y) {
        //xor연산을 해서 서로 다른 비트를 1로 바꿈
        int xorXY = x^y;
        int ans = 0;

        //Brian Kernighan's Algorithm
        //비트가 1인 자리수의 개수를 센다
        while(xorXY > 0){
            xorXY &= (xorXY-1);
            ans++;
        }
        
        return ans;
    }
}